applicationset/generators/pull_request.go

```diff
@@ -214,19 +215,24 @@ func (g *PullRequestGenerator) selectServiceProvider(ctx context.Context, genera
 }
 
 func (g *PullRequestGenerator) github(ctx context.Context, cfg *argoprojiov1alpha1.PullRequestGeneratorGithub, applicationSetInfo *argoprojiov1alpha1.ApplicationSet) (pullrequest.PullRequestService, error) {
+	metricsCtx := &services.MetricsContext{
```

Comment:
I think we should be able at least disable this functionality, i would add new flag in order to control it